hex_dump/views.py

```diff
@@ -226,15 +227,34 @@ def extract_pc_hex(filename):
     return pc_hex_dict
 
 
-def simulate_bash_script(file_name , mtype, ctype, ftype, dtype):
-
-    assemble_cmd = ["riscv32-unknown-elf-gcc",f"-march=rv32i{mtype}{ctype}{ftype}{dtype}", "-mabi=ilp32", "-T", LINKER_SCRIPT, "-static", "-mcmodel=medany", "-fvisibility=hidden", "-nostdlib", "-nostartfiles", "-g", "-o", TMP_ELF, file_name]
+def simulate_bash_script(file_name , mtype, ctype, ftype, dtype , rvtype):
+    extensions = mtype + ctype + ftype + dtype
+    assembletype = 'riscv32'
+    abi = 'ilp32'
+    if rvtype == "rv32":
+        abi = "ilp32"
+        if "f" in extensions and "d" in extensions:
+            abi = "ilp32d"
+        elif "f" in extensions:
+            abi = "ilp32f"
+    else:
+        abi = "lp64"
+        assembletype = 'riscv64'
+        if "f" in extensions and "d" in extensions:
+            abi = "lp64d"
+        elif "f" in extensions:
+            abi = "lp64f"
+    if rvtype == 'rv64':
```

Comment:
This `if` clause seem redundant. If the variable definitions are used elsewhere, it will be better if they are all gathered in the `else` clause for `rv64`